### DISCUSSION TITLE
How to make simple token exchange from 3rd party resource?

### DISCUSSION THREAD

**Discussion by vladrose** [keycloak/keycloak#16387] [Q&A]
Hey all! I have a game on Unreal Engine. It is placed on game launcher like Battle.net with own auth system and you can't start the game without being authenticated, so I can get user's token with some info from the launcher and I want to use keycloack for token exchange to authorize the user on game's server and fetch profile info etc... I don't have enough experience with keycloack and need this feature asap, can somebody help me with simple token exchange flow if I have regular JWT and want to authorize user on my server with it?

I practiced with google OAuth and ran keycloack inside docker locally and it works, but in launcher case I expect some custom logic...

**Comment by victormafe18** [comment]
Hello Vladrose, I have a similar issue now, I have a configured keycloak server for some other issues, and I need to use this keycloak auth in my unreal engine app, but I have no idea on how to setup this, could you show me the way to do this please?

**Reply by vladrose** [reply]
Hey man! No, I used different auth approach. Looks like keycloak can't handle simple token exchange flow
